src/Notifier.cc

```diff
@@ -58,13 +70,20 @@ void Registry::Modified(Modifiable* m) {
 }
 
 void Registry::Terminate() {
-    std::set<Receiver*> receivers;
```

Comment:
I haven't tried, and this might be completely off, but could rewriting the original `Terminate()` as something like the following work without the extra final_receivers? Intuitively by the comments, Triggers should fully unregister themselves and at the end of it registrations be empty (or this results in an endless loop :sweat: )

```
while ( ! registrations.empty() ) {
  const auto& it = registrations.begin();
  it->second->Terminate();
}
```

Reply:
Oooh much better. I've included this in my latest push. (I kept the new `Notifier` API for a `Receiver` to report "hey I'm all done" as it seems reasonable to have that in any case.)

I'm finished with new changes, barring other points to consider. Thanks for the comments!

Reply:
(oh and ASAN confirmed that that change is fine)

Reply:
> (oh and ASAN confirmed that that change is fine)

The ASAN test failed in CI, actually. Let me comment on one `Unref()` where I'm not sure where it went :-)